### PR TITLE
framework/araui: Rendering image with texture uv mapping

### PR DESCRIPTION
**Pull request by gcjjyy**
- framework/araui: Rendering image with texture uv coordinates
- tools/araui: Fix wrong simulator display area

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [2504dbecdc037fae42aae24b7a90b925b1665b19] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2504dbecdc037fae42aae24b7a90b925b1665b19] - Code Rule Check (C++) OK.

File: framework/src/araui/include/ui_widget_internal.h [hunk 1]
```diff
@@ -65,6 +71,11 @@ typedef struct {
 	uint32_t d;
 } tween_info_t;
 
+typedef struct {
+	float u;
+	float v;
+} uv_t;
```
Comment (tyha1115) [inline framework/src/araui/include/ui_widget_internal.h]: This `uv_t` should be `ui_uv_t`, as you use below.

File: framework/src/araui/include/ui_renderer.h [hunk 2]
```diff
@@ -15,17 +15,76 @@
  * language governing permissions and limitations under the License.
  *
  ****************************************************************************/
+#ifndef __UI_RENDERER_H__
+#define __UI_RENDERER_H__
 
-#include "ui_widget_internal.h"
+#include <araui/ui_commons.h>
```
Comment (tyha1115) [inline framework/src/araui/include/ui_renderer.h]: please add `<stdint.h>`

File: framework/src/araui/include/ui_widget_internal.h [hunk 3]
```diff
@@ -29,6 +29,12 @@
 #include "ui_asset_internal.h"
 #include "ui_widget_internal.h"
 #include "ui_commons_internal.h"
+#include "ui_renderer.h"
+
+#define UV_TOP_LEFT     (0)
```
Comment (tyha1115) [inline framework/src/araui/include/ui_widget_internal.h]: How about use `enum`?

File: framework/src/araui/include/ui_widget_internal.h (hunk 1 above)
Comment (gcjjyy) [reply framework/src/araui/include/ui_widget_internal.h]: You're right. I'll fix.

File: framework/src/araui/include/ui_renderer.h (hunk 2 above)
Comment (gcjjyy) [reply framework/src/araui/include/ui_renderer.h]: You're right. I'll fix.

File: framework/src/araui/include/ui_widget_internal.h (hunk 3 above)
Comment (gcjjyy) [reply framework/src/araui/include/ui_widget_internal.h]: You're right. I'll fix.

**Comment by seinfra** [comment]
Target : [25a9b39b46834b39dc175deb977f0f1c543ba5a1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [25a9b39b46834b39dc175deb977f0f1c543ba5a1] - Code Rule Check Result:
framework/src/araui/assets/ui_image_asset.c:151: ERROR: [IDT_M_SCH] switch and case should be at the same indent
framework/src/araui/assets/ui_image_asset.c:163: ERROR: [IDT_M_SCH] switch and case should be at the same indent
framework/src/araui/core/ui_quick_panel.c:71: ERROR: [IDT_M_SCH] switch and case should be at the same indent
framework/src/araui/core/ui_window.c:297: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
framework/src/araui/widgets/ui_button_widget.c:102: ERROR: [IDT_M_SCH] switch and case should be at the same indent
framework/src/araui/widgets/ui_paginator_widget.c:104: ERROR: [IDT_M_SCH] switch and case should be at the same indent
framework/src/araui/widgets/ui_paginator_widget.c:165: ERROR: [IDT_M_SCH] switch and case should be at the same indent
framework/src/araui/widgets/ui_scroll_widget.c:201: ERROR: [IDT_M_SCH] switch and case should be at the same indent

**Comment by seinfra** [comment]
Target : [dc3e8d49c3f17219f21daa8cba852450fd4cb43b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [dc3e8d49c3f17219f21daa8cba852450fd4cb43b] - Code Rule Check Result:
framework/src/araui/widgets/ui_button_widget.c:102: ERROR: [IDT_M_SCH] switch and case should be at the same indent
framework/src/araui/widgets/ui_scroll_widget.c:201: ERROR: [IDT_M_SCH] switch and case should be at the same indent

**Comment by seinfra** [comment]
Target : [339451bd121e7d4c9e5a97e43f1d139b485babcd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [339451bd121e7d4c9e5a97e43f1d139b485babcd] - Code Rule Check OK.

**Pull request merged**